index.js

```diff
@@ -1,87 +1,72 @@
       }
 
-      $beforeUpdate (queryOptions, context) {
-        const maybePromise = super.$beforeUpdate(queryOptions, context)
+      async $beforeUpdate(queryOptions, ...args) {
+        await super.$beforeUpdate(queryOptions, ...args);
 
-        return Promise.resolve(maybePromise).then(() => {
-          if (queryOptions.patch && this[options.passwordField] === undefined) {
-            return
-          }
+        if (queryOptions.patch && this[options.passwordField] === undefined) {
+          return;
+        }
 
-          // hash the password
-          return this.generateHash()
-        })
+        return await this.generateHash();
       }
 
-      /**
-             * Compares a password to a Bcrypt hash
-             * @param  {String}             password  the password...
-             * @return {Promise.<Boolean>}            whether or not the password was verified
-             */
-      verifyPassword (password) {
-        return Bcrypt.compare(password, this[options.passwordField])
+      // Compares a password to a bcrypt hash, returns whether or not the password was verified.
+      async verifyPassword(password) {
+        return await bcrypt.compare(password, this[options.passwordField]);
       }
 
-      /**
-             * Generates a Bcrypt hash
-             * @return {Promise.<(String|void)>}  returns the hash or null
-             */
-      generateHash () {
-        const password = this[options.passwordField]
+      /* Sets the password field to a bcrypt hash of the password.
+       * Only does so if the password is not already a bcrypt hash. */
+      async generateHash() {
+        const password = this[options.passwordField];
 
         if (password) {
           if (this.constructor.isBcryptHash(password)) {
-            throw new Error('bcrypt tried to hash another bcrypt hash')
+            throw new Error('bcrypt tried to hash another bcrypt hash');
           }
 
-          return Bcrypt.hash(password, options.rounds).then((hash) => {
-            this[options.passwordField] = hash
-          })
+          const hash = await bcrypt.hash(password, options.rounds);
+          this[options.passwordField] = hash;
+
+          return hash;
         }
 
-        // throw an error if empty passwords aren't allowed
+        // Throw an error if empty passwords are not allowed.
         if (!options.allowEmptyPassword) {
-          throw new Error('password must not be empty')
+          throw new Error('password must not be empty');
         }
-
-        return Promise.resolve()
       }
 
-      /**
-             * Detect rehashing for avoiding undesired effects
-             * @param {String} str A string to be checked
-             * @return {Boolean} True if the str seems to be a bcrypt hash
-             */
-      static isBcryptHash (str) {
-        return REGEXP.test(str)
+      /* Detect rehashing to avoid undesired effects.
+       * Returns true if the string seems to be a bcrypt hash. */
```

Comment:
I removed some JSDoc comments because I wasn't auto-generating any documentation, but if you want them back, I can add them back.

Reply:
No worries. No need to put them back.